languages/ruby/exercises/concept/hashes/log_line.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module LogLevel
+  UNKNOWN = 0
+  INFO = 1
+  WARNING = 2
+  ERROR = 4
+end
+
+class LogLine
+  LEVEL = {
+    unknown: LogLevel::ERROR,
+    info: LogLevel::INFO,
+    warning: LogLevel::WARNING,
+    error: LogLevel::ERROR
+  }.freeze
+
+  def initialize(line)
+    @line = line
+  end
+
+  def level
+    LEVEL.fetch(level_string.downcase.to_sym, LogLevel::UNKNOWN)
```

Comment:
Here I'm not sure which syntax will be better, the current one or:
``LEVEL[level_string.downcase.to_sym] || LogLevel::UNKNOWN``

Reply:
I think we leave it up the user to choose. I wouldn't mention either approach and just let the student work it out.  But the analyzer could push them towards `fetch` if they use `||`. That would be quite simple to write.

Similarly on the string methods, I'd say "check out the [Ruby docs]... for different methods you can use to manipulate strings" but not explicitely tell them about split etc.

Reply:
I need to have a look at the analyzers I'm don't know how they work 😄